src/app/utils/model.ts

```diff
@@ -0,0 +1,35 @@
+
+import { OpenAI } from "langchain/llms/openai";
+import { Replicate } from "langchain/llms/replicate";
+
+enum LlmModel {
+    OpenAI = 'openai',
+    ReplicateLlama = 'replicate_llama',
+}
+export function getModel() {
```

Comment:
**issue (complexity):** Consider using a configuration map pattern to manage model initialization

The current implementation adds unnecessary complexity by mixing enum usage with string comparisons and scattered configuration. Here's a simpler approach using a model configuration map:

```typescript
const MODEL_CONFIGS = {
  replicate_llama: {
    create: () => new Replicate({
      apiKey: process.env.REPLICATE_API_TOKEN,
      model: "meta/llama13b-v2-chat:2d19859030ff705a87c746f7e96eea03aefb71f166725aee39692f1476566d48",
      input: {
        system_prompt: `You are a helpful assistant, that only communicates using JSON files and no other words.
          The expected output from you has to be: 
          {
              "text": {response}
          }`
      }
    }),
  },
  openai: {
    create: () => new OpenAI({
      modelName: "gpt-3.5-turbo-16k",
      openAIApiKey: process.env.OPENAI_API_KEY,
    }),
  }
};

export function getModel() {
  const modelType = (process.env.LLM_MODEL || "openai").toLowerCase();
  const config = MODEL_CONFIGS[modelType] || MODEL_CONFIGS.openai;

  console.debug(`Using ${modelType}`);
  return config.create();
}
```

This approach:
1. Eliminates the enum+if/else complexity in favor of a simple lookup
2. Groups model configuration with its initialization logic
3. Makes adding new models easier - just add a new entry to MODEL_CONFIGS
4. Maintains the same functionality while reducing code complexity